apps/web/app/(navigation)/MainMenu.tsx

```diff
@@ -0,0 +1,73 @@
+import * as Select from "@radix-ui/react-select";
+import styled from "styled-components";
+import { useRouter } from "next/navigation";
+import { signOut, useSession } from "next-auth/react";
+
+import { useHasScrollBar } from "lib/hooks/useHasScrollBar";
+import { Avatar } from "ui";
+import { useTranslate } from "lib/hooks";
+
+import {
+  SelectContentStyled,
+  SelectItemStyled,
+  SelectTriggerStyled,
+} from "./LanguageSelectorStyled";
+
+export const AvatarStyled = styled(Avatar)`
+  height: 2.1em;
+  width: 2.1em;
+`;
+
+export const MainMenu = () => {
+  const { hasScrollbar } = useHasScrollBar();
+  const router = useRouter();
+  const { data } = useSession();
+  const { t, dict } = useTranslate("MainPage");
+
+  const menuHandler = (value: string) => {
+    if (value === "sign-out") {
+      signOut();
+    }
+    router.push("/");
+  };
```

Comment:
```suggestion
  const menuHandler = (value: string) => {
    value === "sign-out" ? signOut() : "";
  };
  };
```

The signOut function already has redirect function built in thus push will never fire. This can be simplified since you cannot access this menu if you are not logged in already.